tasks/add-fbc-contribution/add-fbc-contribution.yaml

```diff
@@ -77,23 +80,40 @@ spec:
         add_arches=$(jq '.fbc.addArches // []' ${DATA_FILE})
         hotfix=$(jq -r '.fbc.hotfix // "false"' ${DATA_FILE})
         staged_index=$(jq -r '.fbc.stagedIndex // "false"' ${DATA_FILE})
+        product_name=$(jq -r '.fbc.productName // ""' ${DATA_FILE})
+        product_version=$(jq -r '.fbc.productVersion // ""' ${DATA_FILE})
         build_timeout_seconds=$(jq -r --arg build_timeout_seconds ${default_build_timeout_seconds} \
             '.fbc.buildTimeoutSeconds // $build_timeout_seconds' ${DATA_FILE})
-        target_index=$(jq -r '.fbc.targetIndex' ${DATA_FILE})
         fbc_fragment=$(jq -cr '.components[0].containerImage' ${SNAPSHOT_PATH})
 
         timestamp_format=$(jq -r '.fbc.timestampFormat // "%s"' ${DATA_FILE})
         timestamp=$(date "+${timestamp_format}")
 
+        # default tag and targetIndex
+        tag=""
+        target_index=$(params.targetIndex)
+
+        # the targetIndex should be modified based on the hotfix or pre-GA
+        # information the `data.fbc` in release_plan_admission offers
         if [ "${hotfix}" == "true" ]; then
           issue_id=$(jq -r '.fbc.issueId // empty' ${DATA_FILE})
           if [ -z "${issue_id}" ]; then
             echo "Hotfix releases requires the issue id."
             exit 1
           fi
           tag="${issue_id}-${timestamp}"
+        elif [ -n "${product_name}" ]; then
+          if [ -z "${product_version}" ]; then
+            echo "Pre-GA releases requires `fbc.productVersion` set in the ReleasePlanAdmission"
+          fi
+          tag="${product_name}-${product_version}-${timestamp}"
```

Comment:
Oh, is the `target_index` actually the tag that is used? So it will contain the ocp version plus all the extra stuff? In that case isn't it confusing to call part of the tag "index_image" and another part "tag"? Maybe "index_suffix" or something like that would be better? WDYT?

Reply:
Yes, the OCP versions comes from the target_index and it is updated by the `update-ocp-tag` task. So, it merges the tag (already containing the version) with the product info and timestamp.

About changing the name, sure. You are right, it creates a bit of confusion.